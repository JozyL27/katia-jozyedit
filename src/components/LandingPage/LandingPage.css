#lp__main-flex {
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.lp__login {
  position: absolute;
  top: 2vmax;
  right: 2vmax;
}

.lp__header {
  font-size: 2rem;
  margin: 1vmax 0;
}

.lp__logoContainer {
  text-align: center;
}

#lp__container {
  text-align: center;
  padding: 1rem;
}

.lp__about-para {
  width: 80%;
  font-size: 1.5rem;
  margin: 0 auto;
}

.lp__lp-link {
  font-size: 10rem;
  font-weight: bold;
  text-decoration: none;
  background: linear-gradient(to right, rgb(48, 207, 208) 0%, rgb(51, 8, 103) 100%);
  color: #fff;
  background: linear-gradient(-45deg, #EE7752, #E73C7E, rgb(48, 207, 208), rgb(51, 8, 103));
  background-size: 400% 400%;
  position: relative;
  animation: change 10s ease-in-out infinite;
  background-clip: text;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.lp__label {
  font-size: 2.5rem;
}

.lp__input-div {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  margin-top: 2rem;
}

.lp__input {
  width: 17rem;
  border: none;
  border-bottom: .2rem solid black;
  font-size: 1.6rem;
  font-weight: bold;
  text-align: center;
  outline: none;
}

.separator {
  border: .1rem solid lightgrey;
  width: 60%;
  margin: 0 auto;
}

@keyframes change {
  0%{
    background-position: 0 50%;
  }
  50%{
    background-position: 100% 50%;
  }
  100%{
    background-position: 0 50%;
  }
}

@media screen and (min-width: 720px) {
  .lp__header {
    font-size: 2.2rem;
  }

  .lp__about-para {
    font-size: 1.7rem;
    width: 60%;
  }

  .separator {
    width: 50%;
  }
}