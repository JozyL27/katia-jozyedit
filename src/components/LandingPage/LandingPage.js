import React from 'react'
import {Link} from 'react-router-dom'
import './LandingPage.css'

function LandingPage() {
    return (
      <main id='lp__main-flex'>
        <Link to='/login' className='lp__login blue-button'>Login</Link>
        <div id='lp__container'>
          <h1 className='lp__logoContainer'>
          <Link to='/' className='lp__lp-link'>Katia</Link>
          </h1>
          <div className='separator'></div>
          <h2 className='lp__header'>Find new teammates. Squad up.</h2>
          <p className='lp__about-para'>Katia is a social media application made for gamers by gamers. Quickly and easily find others who share the same games and platforms as you.</p>
          <Link to='/signup' className='blue-button blue-button-margin'>Signup</Link>
        </div>
      </main>
    )
}

export default LandingPage
