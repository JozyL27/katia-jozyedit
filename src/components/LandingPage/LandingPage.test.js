import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import LandingPage from './LandingPage'

describe('<LandingPage />', () => {
  // Smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <MemoryRouter>
        <LandingPage />
      </MemoryRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  // Snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <LandingPage />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})