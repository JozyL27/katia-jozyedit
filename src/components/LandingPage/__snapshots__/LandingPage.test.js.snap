// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage /> renders the UI as expected 1`] = `
<main
  id="lp__main-flex"
>
  <a
    className="lp__login blue-button"
    href="/login"
    onClick={[Function]}
  >
    Login
  </a>
  <div
    id="lp__container"
  >
    <h1
      className="lp__logoContainer"
    >
      <a
        className="lp__lp-link"
        href="/"
        onClick={[Function]}
      >
        Katia
      </a>
    </h1>
    <div
      className="separator"
    />
    <h2
      className="lp__header"
    >
      Find new teammates. Squad up.
    </h2>
    <p
      className="lp__about-para"
    >
      Katia is a social media application made for gamers by gamers. Quickly and easily find others who share the same games and platforms as you.
    </p>
    <a
      className="blue-button blue-button-margin"
      href="/signup"
      onClick={[Function]}
    >
      Signup
    </a>
  </div>
</main>
`;
